### PR TITLE
Missing ExtendedErrorInfo fix (#922)

### PR DESCRIPTION
**Pull request by mathewc**
Fix for https://github.com/Azure/azure-webjobs-sdk/issues/922.

See related Azure Storage SDK issue: https://github.com/Azure/azure-storage-net/issues/271

### CODE REVIEW

File: WebJobs.sln [hunk 1]
```diff
@@ -52,6 +52,10 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "WebJobs.Logging", "src\Micr
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "WebJobs.Logging.FunctionalTests", "test\Microsoft.Azure.WebJobs.Logging.FunctionalTests\WebJobs.Logging.FunctionalTests.csproj", "{C8EAAE01-E8CF-4131-9D4B-F0FDF00DA4BE}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Sample", "Sample", "{72A798F0-699B-4C8E-8D43-C1749661471E}"
```
Comment (mathewc) [inline WebJobs.sln]: Been meaning to add a sample app for a while like we have in our other repos. Makes playing around with the code/debugging so much easier.

File: src/Microsoft.Azure.WebJobs.Host/Executors/JobHostContextFactory.cs [hunk 2]
```diff
@@ -82,6 +82,8 @@ public JobHostContextFactory(IStorageAccountProvider storageAccountProvider, ICo
                 hostIdProvider = new DynamicHostIdProvider(storageAccountProvider, () => functionIndexProvider);
             }
 
+            AzureStorageDeploymentValidator.Validate();
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Host/Executors/JobHostContextFactory.cs]: I've verified that this check fails if Microsoft.Data.Services.Client.dll is not present.

**Comment by mathewc** [comment]
@MikeStall Can you take a look at this?

File: src/Microsoft.Azure.WebJobs.Host/AzureStorageDeploymentValidator.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    /// the Storage SDK actually continues to work overall, but some features will silently fail.
+    /// This directly affects our StorageExceptionExtensions which rely on ExtendedErrorInfo being correctly
+    /// populated always. However, when certain assemblies are missing, ExtendedErrorInfo will be null.
+    /// See related issues:
+    /// <see href="https://github.com/Azure/azure-webjobs-sdk/issues/922"/>
+    /// <see href="https://github.com/Azure/azure-storage-net/issues/271"/> 
+    /// </summary>
+    internal static class AzureStorageDeploymentValidator
+    {
+        public static bool Validate()
```
Comment (MikeStall) [inline src/Microsoft.Azure.WebJobs.Host/AzureStorageDeploymentValidator.cs]: Should this just return void? 
And if there's an error, it throws.

Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Host/AzureStorageDeploymentValidator.cs]: Yep - remnants from a refactor :) Will fix,

**Pull request merged**